kmultiversx/src/kmultiversx/kelrond.py

```diff
@@ -0,0 +1,106 @@
+from __future__ import annotations
+
+import sys
+from argparse import ArgumentParser
+from contextlib import contextmanager
+from enum import Enum
+from pathlib import Path
+from tempfile import NamedTemporaryFile
+from typing import TYPE_CHECKING
+
+from pyk.cli.utils import file_path
+from pyk.kdist import kdist
+from pyk.ktool.kprint import KAstOutput, _kast
+from pyk.ktool.krun import _krun
+from pykwasm.scripts.preprocessor import preprocess
+
+if TYPE_CHECKING:
+    from collections.abc import Iterator
+    from subprocess import CompletedProcess
+
+
+class Backend(Enum):
+    LLVM = 'llvm'
+    HASKELL = 'haskell'
+
+
+class MainFile(Enum):
+    MANDOS = 'mandos'
+    KASMER = 'kasmer'
+
+
+def main() -> None:
+    parser = _argument_parser()
+    args, rest = parser.parse_known_args()
+
+    if args.command == 'run':
+        _exec_run(program=args.program, backend=args.backend, main_file=args.main_file)
+    elif args.command == 'kast':
+        _exec_kast(program=args.program, backend=args.backend, main_file=args.main_file, output=args.output)
+
+    raise AssertionError()
+
+
+def _exec_run(*, program: Path, backend: Backend, main_file: MainFile) -> None:
+    definition_dir = kdist.get(f'mx-semantics.{backend.value}-{main_file.value}')
+
+    with _preprocessed(program) as input_file:
+        proc_res = _krun(definition_dir=definition_dir, input_file=input_file, check=False)
+
+    _exit_with_output(proc_res)
```

Comment:
I think, it is better if we get both stderr and stdout of `krun` in the output. Mandos scenarios typically starts with `setExitCode 1` and ends with `setExitCode 0`. When a test fails, the return code is 1, but `stderr` might be empty. In such cases, we would need the final configuration to understand the cause of the failure.

Reply:
Done.